Update Entity-History.md (#4756)
aspnetboilerplate/aspnetboilerplate: doc/WebSite/Entity-History.md

The Use method returns an IDisposable and it **must be disposed**. Once the return
value is disposed, the Reason is automatically restored to the previous value.

> 

Reviewer: ```suggestion
```

### IEntitySnapshotManager

You may need to get snapshot of your entity on a date. To get snapshot of your entity you can use `IEntitySnapshotManager.GetSnapshotAsync(...)` method as shown below.

```csharp
  public class MyService
    {
        private readonly IEntitySnapshotManager _entitySnapshotManager;

        public MyService(IEntitySnapshotManager entitySnapshotManager)
        {
            _entitySnapshotManager = entitySnapshotManager;
        }

        public Task<EntityHistorySnapshot> GetMySnapshot(long id,DateTime time)
        {
            return _entitySnapshotManager.GetSnapshotAsync<MyEntity, long>(id,time);
        }
      
        public async Task<string> GetMyProperty(long id,DateTime time)
        {

>            var snapshot = await GetMySnapshot(id, time);

Reviewer: ```suggestion
           var snapshot = await GetMyEntitySnapshot(id, time);
```

           if(snapshot.IsPropertyChanged(nameof(MyEntity.MyProperty)))
           {

>                var stacktree = snapshot.PropertyChangesStackTree[nameof(MyEntity.MyProperty)];

Reviewer: ```suggestion
               // var stacktree = snapshot.PropertyChangesStackTree[nameof(MyEntity.MyProperty)];
```

               return snapshot[nameof(MyEntity.MyProperty)];               
           }
           return null;
        }
    }
```

> It returns `EntityHistorySnapshot` which contains all snapshot on the given date and stack tree of them

Reviewer: ```suggestion
`IEntitySnapshotManager.GetSnapshotAsync(...)` returns an `EntityHistorySnapshot` that contains the snapshot value of all changed properties on the given date and a stack tree of the changes.

```
